servers/rendering/rendering_device_driver.h

```diff
@@ -468,6 +468,7 @@ class RenderingDeviceDriver : public RenderingDeviceCommons {
 	// Only meaningful if API_TRAIT_SHADER_CHANGE_INVALIDATION is SHADER_CHANGE_INVALIDATION_ALL_OR_NONE_ACCORDING_TO_LAYOUT_HASH.
 	virtual uint32_t shader_get_layout_hash(ShaderID p_shader) { return 0; }
 	virtual void shader_free(ShaderID p_shader) = 0;
+	virtual void shader_destroy_modules(ShaderID p_shader) = 0;
```

Comment:
Imp mentation is missing from DX12 driver (or it can be changed to non-abstract dummy here).